### PR TITLE
OO(oooh) refactor

### PR DESCRIPTION
**Pull request by brendt**
@freekmurze I decided I wanted you to review this, because I'm pretty happy with how it turned out and wanted you to see it in its whole. (I'm using spatie/typed :D)

Addresses #5

### CODE REVIEW

File: src/Parser/FileParser.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    protected $path;
+
+    public function __construct(string $path)
+    {
+        $this->path = $path;
+    }
+
+    public function getParsed(): Page
+    {
+        $contents = file_get_contents($this->path);
```
Comment (peter279k) [inline src/Parser/FileParser.php:26]: Does this file path need to check whether it's existed?
How about using the `file_exists` function to do this?

File: src/Parser/FileParser.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+        $lines = explode(PHP_EOL, $contents);
+
+        $page = new Page();
+
+        foreach ($lines as $line) {
+            if (!strlen($line)) {
```
Comment (peter279k) [inline src/Parser/FileParser.php]: I think using the `=== 0` is the proper way.

File: src/Parser/FileParser.php (hunk 1 above)
Comment (brendt) [reply src/Parser/FileParser.php:26]: I don't want this part of the code to concern itself with that kind of high level error handling. Whether the file exists or not should be done on a higher level. I've added the check when constructing the `FileParser`.

File: src/Parser/FileParser.php (hunk 2 above)
Comment (brendt) [reply src/Parser/FileParser.php]: You're right, thanks!

File: src/Parser/FileParser.php (hunk 1 above)
Comment (peter279k) [reply src/Parser/FileParser.php:26]: Yes. The proper way is to check whether the file is existed in `FileParser` class constructor :+1: .

**Comment by peter279k** [comment]
@brendt, is this PR ready to be merged?

Thanks.

**Comment by brendt** [comment]
It is, I was away on holiday. Merging it now!

**Comment by brendt** [comment]
It's tagged here: https://github.com/spatie/code-outliner/releases/tag/0.1.3

**Pull request merged**